packages/runtime/container-runtime/src/summary/documentSchema.ts

```diff
@@ -601,55 +601,73 @@ export class DocumentsSchemaController {
 	 * @param local - whether op is local
 	 * @param sequenceNumber - sequence number of the op
 	 * @returns - true if schema was accepted, otherwise false (rejected due to failed CAS)
+	 * @deprecated - it has been replaced by processDocumentSchemaMessages instead.
 	 */
 	public processDocumentSchemaOp(
 		content: IDocumentSchemaChangeMessage,
 		local: boolean,
 		sequenceNumber: number,
 	) {
-		this.validateSeqNumber(content.refSeq, this.documentSchema.refSeq, "content.refSeq");
-		this.validateSeqNumber(this.documentSchema.refSeq, sequenceNumber, "refSeq");
-		// validate is strickly less, not equal
-		assert(
-			this.documentSchema.refSeq < sequenceNumber,
-			0x950 /* time should move forward only! */,
-		);
-
-		if (content.refSeq !== this.documentSchema.refSeq) {
-			// CAS failed
-			return false;
-		}
-
-		// This assert should be after checking for successful CAS above.
-		// This will ensure we do not trip on our own messages that are no longer wanted as we processed someone else schema change message.
-		assert(
-			!local || (this.explicitSchemaControl && this.futureSchema !== undefined),
-			0x951 /* not sending ops */,
-		);
+		return this.processDocumentSchemaMessages([content], local, sequenceNumber);
+	}
 
-		// Changes are in effect. Immediately check that this client understands these changes
-		checkRuntimeCompatibility(content, "change");
+	/**
+	 * Process document schema change messages
+	 * Called by ContainerRuntime whenever it sees document schema messages.
+	 * @param contents - contents of the messages
+	 * @param local - whether op is local
+	 * @param sequenceNumber - sequence number of the op
+	 * @returns - true if schema was accepted, otherwise false (rejected due to failed CAS)
+	 */
+	public processDocumentSchemaMessages(
```

Comment:
Note: this class isn't marked as `@sealed`, so adding a new public member is a breaking change. Presumably this needs to wait for the next legacy breaking changes release.

Reply:
Longer term: do we need to export this class for external use? Or could we refactor and just export an interface?

If we do need to export this, do we expect external users to implement/extend it? If not, it would probably be good to mark this as `@sealed`.

Reply:
These are all helper classes for container runtime. They should all be internal AFAIK. No one should be implementing or extending it. The only reason it is alpha / legacy is because it is used in Container Runtime's constructor.

Reply:
Maked it @sealed. @vladsud FYI, in case you think it should not be sealed.